pkg/controller/mcmhub/ansiblejob.go

```diff
@@ -95,7 +96,7 @@ func (jIns *JobInstances) registryJobs(gClt GitOps, subIns *subv1.Subscription,
 		logger.Info(fmt.Sprintf("registered ansiblejob %s", nxKey))
 
 		if _, ok := jobRecords.InstanceSet[nxKey]; !ok {
-			if forceRegister && len(jobRecords.Instance) > 0 {
+			if !commitIDChanged && placementDecisionUpdated && len(jobRecords.Instance) > 0 {
```

Comment:
Can you add some logs here so when this is not run, we know from the logs?

Reply:
added some more log please see new commit.